webapp/sources/rudder/rudder-rest/src/main/scala/com/normation/rudder/rest/lift/NodeApi.scala

```diff
@@ -783,10 +791,16 @@ class NodeApiService8 (
       nodes <- nodeInfoService.getAll() ?~! s"Could not find nodes informations"
 
     } yield {
+      // remote run only works for CFEngine based agent
+      val okNodes = nodes.toList.flatMap {
+        case (_, node) if(node.agentsName.exists(a => a.agentType == AgentType.CfeEnterprise || a.agentType == AgentType.CfeCommunity)) =>
+          Some(node)
+        case _ => None
+      }
 
       val res =
       for {
-        node <- nodes.values.toList
+        node <- okNodes
```

Comment:
I would have kept the whole list here but instead  in the result for non compatible nodes i would have changed the result ro "Skipped, agent is not compatible with remote-run" or just "skipped"

Reply:
good idea